ui/app/appointments/directives/datePicker.js

```diff
@@ -22,15 +22,23 @@ angular.module('bahmni.appointments')
                 $scope.viewDate = $scope.viewDate && dateUtil.addDays($scope.viewDate, 1);
             };
 
-            $scope.$watch("viewDate", $scope.onChange($scope.viewDate));
+            $scope.$watch("viewDate", function (viewDate) {
+                if (!viewDate) {
+                    $scope.viewDate = $scope.lastValidDate;
```

Comment:
Is there never a case where `$scope.lastValidDate` is not yet defined here?

Isn't it necessary to at least initialize `$scope.lastValidDate` as today?